### PR TITLE
feat: 🎸 add recordEvent to orchestration and command queue

### LINKED ISSUE #139: Add log level support
We've found that `recordError` is a useful replacement for custom client error logging. However to completely replace what we have now we'd need support for other log levels, e.g. info, warn.

This could be provided by a `recordLog` or `recordEvent` command that supports arbitrary logging. Is this something that would be considered in scope for the rum client?

### PR DESCRIPTION
**Pull request by maniator**
Add `recordEvent` to RUM which will be added to the `interaction` telemetry on page load.

Closes #139 

---

Example JSON sent via [PutRumEvents](https://docs.aws.amazon.com/cloudwatchrum/latest/APIReference/API_PutRumEvents.html) api (in an integ test):

```
{
  BatchId: 'b18cb16b-47f8-4d16-a67d-831e1f17ae3c',
  AppMonitorDetails: { id: 'abc123', version: '1.0' },
  UserDetails: {
    userId: 'e77ad812-00cd-4fdb-a051-a247e5618a7b',
    sessionId: '9cd9bec8-c5f7-4e9b-a8ae-a75fad02f749'
  },
  RumEvents: [
    {
      id: 'bad26fe3-e763-4429-87de-df43b49cf917',
      timestamp: 1658867045,
      type: 'com.amazon.rum.js_general_event',
      metadata: '{"version":"1.0.0","browserLanguage":"en-US","browserName":"Chrome Headless","browserVersion":"103.0.5060.134","osName":"Mac OS","osVersion":"10.15.7","deviceType":"desktop","platformType":"web","domain":"localhost","title":"RUM Integ Test","pageId":"/js_general_event.html","interaction":0}',
      details: '{"version":"1.0.0","name":"random-event","data":"Some random event"}'
    }
  ]
}
```

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by maniator** [comment]
Fixing up integ tests now.


False pass in actions atm

File: src/orchestration/Orchestration.ts [hunk 1]
```diff
@@ -473,6 +488,7 @@ export class Orchestration {
             },
             [TelemetryEnum.Interaction]: (config: object): InternalPlugin[] => {
                 return [new DomEventPlugin(config)];
+                return [new JsEventPlugin(config)];
```
Comment (jsutter909) [inline src/orchestration/Orchestration.ts:491]: Should be a list of both plugins, not two return statements.

File: src/orchestration/Orchestration.ts [hunk 2]
```diff
@@ -341,6 +348,14 @@ export class Orchestration {
         this.pluginManager.record(JS_ERROR_EVENT_PLUGIN_ID, error);
     }
 
+    /**
+     * Record an error using the JS event plugin.
```
Comment (jsutter909) [inline src/orchestration/Orchestration.ts:352]: Record an `event`

**Review by jsutter909 — COMMENTED** [review]
I have no real comments on the logic. There are some typos to review though. Otherwise looks great.

File: src/orchestration/Orchestration.ts (hunk 1 above)
Comment (maniator) [reply src/orchestration/Orchestration.ts:491]: you are correct :-)

**Comment by maniator** [comment]
> I have no real comments on the logic. There are some typos to review though. Otherwise looks great.

Resolved 😄

**Pull request closed without merging**